src/chaos-experiments/models/chaosExperimentsManager.js

```diff
@@ -0,0 +1,192 @@
+};
+
+const buildExperimentResource = (kubernetesChaosConfig, jobId) => {
+    const labels = {
+        [CHAOS_EXPERIMENT_LABELS.APP]: PREDATOR_RUNNER_PREFIX,
+        [CHAOS_EXPERIMENT_LABELS.JOB_ID]: jobId
+    };
+    const { metadata = {} } = kubernetesChaosConfig;
+    return {
+        ...kubernetesChaosConfig,
+        metadata: {
+            ...metadata,
+            labels: {
+                ...metadata.labels,
+                ...labels
+            }
+        }
+    };
+};
+
+const getFutureJobExperiments = async function (timestamp, contextId) {
+    return databaseConnector.getFutureJobExperiments(timestamp, contextId);
+};
+
+const reloadSingleChaosExperiment = async function (futureJobExperiment, timestamp){
+    try {
+        const calculatedStartAfter = futureJobExperiment.start_time - timestamp;
+        const chaosExperiment = await getChaosExperimentById(futureJobExperiment.experiment_id);
+        jobExperimentHandler.scheduleChaosExperiment(chaosExperiment.kubeObject, futureJobExperiment.job_id, futureJobExperiment.id, calculatedStartAfter);
+    } catch (error) {
+        throw new Error('Unable to reload job experiments ' + futureJobExperiment.id + ' , error: ' + error);
+    }
+};
+
+const reloadChaosExperiments = module.exports.reloadChaosExperiments = async function() {
+    const contextId = httpContext.get(CONTEXT_ID);
+    try {
+        const timestamp = Date.now();
+        const futureJobExperiments = await getFutureJobExperiments(timestamp, contextId);
+        for (const futureJobExperiment of futureJobExperiments) {
+            await reloadSingleChaosExperiment(futureJobExperiment, timestamp);
+        }
+    } catch (error) {
+        throw new Error('Unable to reload job experiments , error: ' + error);
+    }
+};
+
+const setPlatform = module.exports.setPlatform = async function () {
+    const jobPlatform = await configHandler.getConfigValue(CONFIG.JOB_PLATFORM);
+    if (jobPlatform.toUpperCase() !== KUBERNETES) return;
+    const platform = jobPlatform.toLowerCase();
+    connector = require(`./${platform}/chaosExperimentConnector`);
+    jobExperimentHandler = require(`./../../jobs/models/${platform}/jobExperimentsHandler`);
+    return jobPlatform;
+};
+
+module.exports.init = async function () {
+    const platform = await setPlatform();
+    if (!platform) return;
+    await reloadChaosExperiments();
+};
+
+const stopResourcesOfJobIdAndExperiment = async (jobId, kind, namespace) => {
+    try {
+        await connector.deleteAllResourcesOfKindAndJob(kind, namespace, jobId);
+    } catch (e){
+        logger.error(`Failed to get resources of kind ${kind} of jobId ${jobId}: ${e}`);
+    }
+};
+
+module.exports.stopJobExperimentsByJobId = async function(jobId) {
+    try {
+        const jobExperiments = await getChaosJobExperimentsByJobId(jobId);
+        const now = Date.now();
+        const relevantJobExperiments = jobExperiments.filter(experiment => experiment.start_time <= now);
```

Comment:
instead of filtering relevant expirements that started, i'd just try and delete all to make sure we don't miss anything, also would simplify this function

Reply:
did it thanks